common/src/test/scala/objects/BuildingTest.scala

```diff
@@ -0,0 +1,169 @@
+      val veh = Vehicle(GlobalDefinitions.quadstealth)
+
+      ao.Owner = veh
+      ao.Owner mustEqual veh
+    }
+
+    "not be owned by an unexpected object" in {
+      val ao = new AmenityObject()
+      //ao.Owner = net.psforever.objects.serverobject.mblocker.Locker() //will not compile
+      ok
+    }
+
+    "confer faction allegiance through ownership" in {
+      //see FactionAffinityTest
+      val ao = new AmenityObject()
+      val bldg = Building(10, Zone.Nowhere)
+      ao.Owner = bldg
+      bldg.Faction mustEqual PlanetSideEmpire.NEUTRAL
+      ao.Faction mustEqual PlanetSideEmpire.NEUTRAL
+
+      bldg.Faction = PlanetSideEmpire.TR
+      bldg.Faction mustEqual PlanetSideEmpire.TR
+      ao.Faction mustEqual PlanetSideEmpire.TR
+    }
+  }
+}
+
+class BuildingTest extends Specification {
+  "Building" should {
+    "construct" in {
+      val bldg = Building(10, Zone.Nowhere)
+      bldg.Id mustEqual 10
+      bldg.Actor mustEqual ActorRef.noSender
+      bldg.Amenities mustEqual Nil
+      bldg.Zone mustEqual Zone.Nowhere
+      bldg.Faction mustEqual PlanetSideEmpire.NEUTRAL
+    }
+
+    "change faction affinity" in {
+      val bldg = Building(10, Zone.Nowhere)
+      bldg.Faction mustEqual PlanetSideEmpire.NEUTRAL
+
+      bldg.Faction = PlanetSideEmpire.TR
+      bldg.Faction mustEqual PlanetSideEmpire.TR
+    }
+
+    "keep track of amenities" in {
+      val bldg = Building(10, Zone.Nowhere)
+      val door1 = Door(GlobalDefinitions.door)
+      val door2 = Door(GlobalDefinitions.door)
+
+      bldg.Amenities mustEqual Nil
+      bldg.Amenities = door2
+      bldg.Amenities mustEqual List(door2)
+      bldg.Amenities = door1
+      bldg.Amenities mustEqual List(door2, door1)
+      door1.Owner mustEqual bldg
+      door2.Owner mustEqual bldg
+    }
+  }
+}
+
+class BuildingControl1Test extends ActorTest {
+  "Building Control" should {
+    "construct" in {
+      val bldg = Building(10, Zone.Nowhere)
+      bldg.Actor = system.actorOf(Props(classOf[BuildingControl], bldg), "test")
+      assert(bldg.Actor != ActorRef.noSender)
+    }
+  }
+}
+
+class BuildingControl2Test extends ActorTest {
+  "Building Control" should {
+    "convert and assert faction affinity on convert request" in {
+      val bldg = Building(10, Zone.Nowhere)
+      bldg.Faction = PlanetSideEmpire.TR
+      bldg.Actor = system.actorOf(Props(classOf[BuildingControl], bldg), "test")
+      assert(bldg.Faction == PlanetSideEmpire.TR)
+
+      bldg.Actor ! FactionAffinity.ConvertFactionAffinity(PlanetSideEmpire.VS)
+      val reply = receiveOne(Duration.create(100, "ms"))
+      assert(reply.isInstanceOf[FactionAffinity.AssertFactionAffinity])
+      assert(reply.asInstanceOf[FactionAffinity.AssertFactionAffinity].obj == bldg)
+      assert(reply.asInstanceOf[FactionAffinity.AssertFactionAffinity].faction == PlanetSideEmpire.VS)
+      assert(bldg.Faction == PlanetSideEmpire.VS)
+    }
+  }
+}
+
+class BuildingControl3Test extends ActorTest {
```

Comment:
You get a gold star for these tests. This is great, really helps me understand how amenities work!